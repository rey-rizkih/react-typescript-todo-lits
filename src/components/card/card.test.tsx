import { render, screen, cleanup } from "@testing-library/react";
import Card from ".";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Renders in window", () => {
  render(<Card title="Test"></Card>);
  const card = screen.getByTestId("card");

  expect(card).toBeInTheDocument();
});

it("Renders button correctly", () => {
  render(<Card title="Test"></Card>);
  const card = screen.getByTestId("card");

  expect(card).toHaveTextContent("Test");
});

it("Matches snapshot", () => {
  const tree = renderer.create(<Card title="Test"></Card>).toJSON();
  expect(tree).toMatchSnapshot();
});
