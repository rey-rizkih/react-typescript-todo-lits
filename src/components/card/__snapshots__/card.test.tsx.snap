// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="MuiBox-root css-1ssq9ig"
  data-testid="card"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
  >
    Test
  </h2>
</div>
`;
